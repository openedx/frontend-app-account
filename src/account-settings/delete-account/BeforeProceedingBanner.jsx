import { faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { FormattedMessage, getSiteConfig, useIntl } from '@openedx/frontend-base';
import { Hyperlink } from '@openedx/paragon';
import PropTypes from 'prop-types';

// Messages
import messages from './messages';

// Components
import Alert from '../Alert';

const BeforeProceedingBanner = (props) => {
  const { instructionMessageId, supportArticleUrl } = props;
  const intl = useIntl();

  return (
    <Alert
      className="alert-warning mt-n2"
      icon={<FontAwesomeIcon className="mr-2" icon={faExclamationTriangle} />}
    >
      <FormattedMessage
        id="account.settings.delete.account.before.proceeding"
        defaultMessage="Before proceeding, please {actionLink}."
        description="Error that appears if you are trying to delete your account, but something about your account needs attention first.  The actionLink will be instructions, such as 'unlink your Facebook account'."
        values={{
          actionLink: supportArticleUrl ? (
            <Hyperlink destination={supportArticleUrl}>
              {intl.formatMessage(messages[instructionMessageId])}
            </Hyperlink>
          ) : (
            intl.formatMessage(messages[instructionMessageId])
          ),
          siteName: getSiteConfig().siteName,
        }}
      />
    </Alert>
  );
};

BeforeProceedingBanner.propTypes = {
  instructionMessageId: PropTypes.string.isRequired,
  supportArticleUrl: PropTypes.string.isRequired,
};

export default BeforeProceedingBanner;
