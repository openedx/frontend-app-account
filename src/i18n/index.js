// Placeholder be overridden by `make pull_translations`
export default {
  ar: {},
  'zh-hk': {},
  'zh-cn': {},
  uk: {},
  'tr-tr': {},
  th: {},
  te: {},
  ru: {},
  'pt-pt': {},
  'pt-br': {},
  'it-it': {},
  id: {},
  hi: {},
  he: {},
  'fr-ca': {},
  fa: {},
  'es-es': {},
  'es-419': {},
  el: {},
  'de-de': {},
  da: {},
  bo: {},
};
