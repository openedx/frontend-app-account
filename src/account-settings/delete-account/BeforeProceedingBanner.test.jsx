import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider, injectIntl, createIntl } from '@edx/frontend-platform/i18n';

jest.mock('react-dom', () => ({
  ...jest.requireActual('react-dom'),
  createPortal: jest.fn(node => node), // Mock portal behavior
}));

import BeforeProceedingBanner from './BeforeProceedingBanner'; // eslint-disable-line import/first

const IntlBeforeProceedingBanner = injectIntl(BeforeProceedingBanner);

describe('BeforeProceedingBanner', () => {
  it('should match the snapshot if SUPPORT_URL_TO_UNLINK_SOCIAL_MEDIA_ACCOUNT does not have a support link', () => {
    const props = {
      instructionMessageId: 'account.settings.delete.account.please.unlink',
      intl: createIntl({ locale: 'en' }),
      supportArticleUrl: '',
    };
    const tree = renderer
      .create((
        <IntlProvider locale="en">
          <IntlBeforeProceedingBanner
            {...props}
          />
        </IntlProvider>
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match the snapshot when SUPPORT_URL_TO_UNLINK_SOCIAL_MEDIA_ACCOUNT has a support link', () => {
    const props = {
      instructionMessageId: 'account.settings.delete.account.please.unlink',
      intl: createIntl({ locale: 'en' }),
      supportArticleUrl: 'http://test-support.edx',
    };
    const tree = renderer
      .create((
        <IntlProvider locale="en">
          <IntlBeforeProceedingBanner
            {...props}
          />
        </IntlProvider>
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
