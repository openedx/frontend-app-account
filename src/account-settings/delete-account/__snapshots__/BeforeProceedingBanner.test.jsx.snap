// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeforeProceedingBanner should match the snapshot if SUPPORT_URL_TO_UNLINK_SOCIAL_MEDIA_ACCOUNT does not have a support link 1`] = `
<div
  className="alert d-flex align-items-start alert-warning mt-n2"
>
  <div>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-triangle-exclamation mr-2"
      data-icon="triangle-exclamation"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7 .2 40.1S486.3 480 472 480L40 480c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8 .2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24l0 112c0 13.3 10.7 24 24 24s24-10.7 24-24l0-112c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0 -64 0 32 32 0 1 0 64 0z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </div>
  <div>
    Before proceeding, please unlink all social media accounts.
  </div>
</div>
`;

exports[`BeforeProceedingBanner should match the snapshot when SUPPORT_URL_TO_UNLINK_SOCIAL_MEDIA_ACCOUNT has a support link 1`] = `
<div
  className="alert d-flex align-items-start alert-warning mt-n2"
>
  <div>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-triangle-exclamation mr-2"
      data-icon="triangle-exclamation"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7 .2 40.1S486.3 480 472 480L40 480c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8 .2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24l0 112c0 13.3 10.7 24 24 24s24-10.7 24-24l0-112c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0 -64 0 32 32 0 1 0 64 0z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </div>
  <div>
    Before proceeding, please 
    <a
      className="pgn__hyperlink default-link standalone-link"
      href="http://test-support.edx"
      target="_self"
    >
      unlink all social media accounts
    </a>
    .
  </div>
</div>
`;
